### PR TITLE
[DataGrid] Remove deprecated `components` and `componentsProps`

### LINKED ISSUE #7940: Remove deprecated `components` and `componentsProps` props

### PR DESCRIPTION
**Pull request by MBilalShafi**
Closes #7940 

TODOs:
- [x] Fix slots descriptions in API docs
- [x] Update migration guide
- [ ] Make codemod part of `preset-safe` (separate PR)

### CODE REVIEW

File: docs/scripts/api/buildComponentsDocumentation.ts [hunk 1]
```diff
@@ -134,10 +134,10 @@ function extractSlots(options: {
...
-    // Workaround to generate correct (camelCase) keys for slots in v6 `API Reference` documentation
-    // TODO v7: Remove camelCase once `Grid(Pro|Premium)SlotsComponent` type is refactored to have `camelCase` names
-    // Shifting to `slots` prop instead of `components` prop strips off the `default` property due to deduced type `UncapitalizedGridSlotsComponent`
-    const slotName = camelCase(name);
+    // Workaround to generate correct (camelCase) keys for slots in `API Reference` documentation
+    // TODO v7: Remove camelCase from pickers once the `components` prop is removed
+    // Shifting to `slots` prop instead of `components` prop strips off the `default` property due to deduced type `UncapitalizedSlotsComponent`
+    const slotName = project.name.includes('grid') ? name : camelCase(name);
 
     slots[slotName] = {
```
Comment (MBilalShafi) [inline docs/scripts/api/buildComponentsDocumentation.ts:142]: @mui/explore we might need to do this once pickers also stop supporting `components/componentsProps`
```suggestion
    slots[name] = {
```

Comment (alexfauquette) [reply docs/scripts/api/buildComponentsDocumentation.ts:142]: It's ready #10700

File: docs/data/migration/migration-data-grid-v6/migration-data-grid-v6.md [hunk 2]
```diff
@@ -77,9 +77,9 @@ Below are described the steps you need to make to migrate from v6 to v7.
 
 - -->
 
-<!-- ### Removed props
+### Removed props
 
-- -->
+- Props `components` and `componentsProps` have been removed. Use `slots` and `slotProps` instead. See [components section](/x/react-data-grid/components/) for more details.
```
Comment (cherniavskii) [inline docs/data/migration/migration-data-grid-v6/migration-data-grid-v6.md:82]:
```suggestion
- The deprecated props `components` and `componentsProps` have been removed. Use `slots` and `slotProps` instead. See [components section](/x/react-data-grid/components/) for more details.
```

**Review by cherniavskii — APPROVED** [review]

File: docs/scripts/api/buildComponentsDocumentation.ts (hunk 1 above)
Comment (MBilalShafi) [reply docs/scripts/api/buildComponentsDocumentation.ts:142]: I merged this one, kindly consider merging the latest `next` and doing the diff mentioned in https://github.com/mui/mui-x/pull/10911#discussion_r1386277625 in the pickers' PR

**Pull request merged**